### ISSUE TITLE
'Manage Company' and 'Company Selector' not working for Company Managers of Child Companies in Moodle 4

### ISSUE THREAD

**Issue by mikexis** [iomad/iomad#1756]
When choosing 'Manage Company' as a Company Manager in Moodle 4, it redirects to the IOMAD Dashboard or My Courses depended on where the IOMAD admin block is situated, it then does not select/change the company to be managed.   The Company Selector also does not show at all for Company Managers with Child Companies.  Tried rolling back to IOMAD 3.10 and it works fine, but not in 4.06.

**Comment by mikexis** [comment]
Changing line 264 in \blocks\iomad_company_admin\index.phop

From `if (!iomad::has_capability('block/iomad_company_admin:company_view_all', $systemcontext)) {`

To `if (!iomad::has_capability('block/iomad_company_admin:company_view', $systemcontext)) {`

Resolves the selector issue, but the 'Manage Company' under 'Manage Companies' section still doesn't work, but at least child companies can be selected.

**Comment by mikexis** [comment]
To update: this only occurs when the Dashboard is hidden or the Iomad Admin block is not on the dashboard.

**Comment by Telford0** [comment]
Trading crypto and making profit, an expert like Lauren B Alex will guide you to make good profits,  she is on Facebook.

**Comment by turf212** [comment]
The lines you've changed in code are to check if you can see all of the companies - the else part of that checks to see which companies the user is allocated to and then displays the selector for that if there are more than one.

If the users is also getting the redirect it sounds like they aren't in any company.  How did you allocate the company manager?  It should have been (from within the IOMAD Dashboard):

Select company
Create user and either assign them the company manager role on creation
or Edit users -> turn on editing -> change role to company manager
of Department users and managers -> select company manager role and allocate them to the role at the top of the company.

If you've used the Moodle Site admin tools then it's not associated the user to any company - which is what I suspect has happened.

**Comment by mikexis** [comment]
Thanks for the response Derick,  I have just tested the scenario on the iomad demo site and the experience is the same.  When I assign the testadmin as Company Manager of two companies they have no company selector.    Scenario tested   'Edit users -> turn on editing -> change role to company manager of Department users and managers -> select company manager role and allocate them to the role at the top of the company.'

The issue with the Manage Companies links not working is also the same on the demo site, if the dashboard is hidden (or the Company Admin block on dashboard is hidded/removed), Manage Companies stops working even as a Site Admin - I found the fix for the latter is to change /my to /blocks/iomad_company_admin in the new company admin index.html.
